Sources/ViewRecycler.swift

```diff
@@ -96,23 +107,36 @@ class ViewRecycler {
         }
         viewsById.removeAll()
 
-        for view in unidentifiedViews where view.isLayoutKitView {
+        for view in unidentifiedViews where view.type == .managed {
             view.removeFromSuperview()
         }
         unidentifiedViews.removeAll()
     }
+
+    func markViewsAsRoot(_ views: [View]) {
+        views.forEach { $0.type = .root }
+    }
 }
 
 private var viewReuseIdKey: UInt8 = 0
-private var isLayoutKitViewKey: UInt8 = 0
+private var typeKey: UInt8 = 0
 
 extension View {
 
+    enum ViewType: UInt8 {
+        // Indicates the view was not created by LayoutKit and should not be modified.
+        case unmanaged
+        // Indicates the view is managed by LayoutKit that can be safely removed.
+        case managed
+        // Indicates the view is managed by LayoutKit but was generated by another layout and should not be modified.
+        case root
+    }
+
     /// Calls visitor for each transitive subview.
-    func walkSubviews(visitor: (View) -> Void) {
-        for subview in subviews {
+    func walkNonRootSubviews(visitor: (View) -> Void) {
+        for subview in subviews where subview.type != .root {
```

Comment:
Is it necessary to walk unmanaged subviews? Won't the subviews of the subview be either unmanaged or root ones?

Reply:
This is a good point. Let's tackle it as a separate change.